packages/blade/src/components/Amount/Amount.tsx

```diff
@@ -0,0 +1,264 @@
+import type { ReactElement } from 'react';
+import React from 'react';
+import type { HeadingProps, TextProps, TitleProps } from '../Typography';
+import {
+  amountFontSizes,
+  currencyAbbreviationsMapping,
+  currencyLocaleMapping,
+  currencyPrefixMapping,
+  affixFontSizes,
+} from './amountTokens';
+import { BaseText } from '~components/Typography/BaseText';
+import type { Feedback } from '~tokens/theme/theme';
+import type { BaseTextProps } from '~components/Typography/BaseText/types';
+import BaseBox from '~components/Box/BaseBox';
+import type { TestID } from '~src/_helpers/types';
+import { getPlatformType, metaAttribute, MetaConstants } from '~utils';
+import { getStyledProps } from '~components/Box/styledProps';
+import type { StyledPropsBlade } from '~components/Box/styledProps';
+
+type Currency = 'INR' | 'MYR';
+
+type AmountProps = {
+  /**
+   * The value to be rendered within the component.
+   *
+   */
+  value: number;
+  /**
+   * Sets the intent of the amount.
+   *
+   * @default undefined
+   */
+  intent?: Exclude<Feedback, 'neutral'>;
+  /**
+   * Sets the size of the amount.
+   *
+   * @default 'body-medium'
+   */
+  size?:
```

Comment:
Optional: I feel we can maintain a union separately since there are limited options, seems too complex to read. eg. `title-small | title-medium | ...`